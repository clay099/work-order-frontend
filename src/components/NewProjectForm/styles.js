import { makeStyles } from "@material-ui/core/styles";
import { paper, form, submit } from "../../styles/styleObjects";

const useStyles = makeStyles((theme) => {
	let paperObj = paper(theme);
	let formObj = form(theme);
	let submitObj = submit(theme);
	return { ...paperObj, ...formObj, ...submitObj };
});

export default useStyles;
