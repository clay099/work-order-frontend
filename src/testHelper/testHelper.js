// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
