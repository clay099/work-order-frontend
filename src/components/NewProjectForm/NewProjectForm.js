import React from "react";
import useFields from "../../hooks/useFields";
import { useDispatch } from "react-redux";
import { useHistory } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";
import { Button, Container, CssBaseline, Typography } from "@material-ui/core";
import BasicProjectForm from "../BasicProjectForm/BasicProjectForm";
import { createNewProjectWithAPI } from "../../actions/projects";

const useStyles = makeStyles((theme) => ({
	paper: {
		marginTop: theme.spacing(2),
		display: "flex",
		flexDirection: "column",
		alignItems: "center",
	},
	form: {
		width: "100%", // Fix IE 11 issue.
		marginTop: theme.spacing(3),
	},
	submit: {
		margin: theme.spacing(3, 0, 2),
	},
}));

const NewProjectForm = ({ token }) => {
	const classes = useStyles();
	let INITIALSTATE = {
		firstName: "",
		lastName: "",
		email: "",
		phone: "",
		password: "",
	};

	const { formData, handleChange, resetFormData } = useFields(INITIALSTATE);

	const dispatch = useDispatch();
	const history = useHistory();

	const handleSubmit = async (e) => {
		e.preventDefault();
		let resp = await dispatch(
			createNewProjectWithAPI({
				description: formData.description,
				streetAddress: formData.streetAddress,
				zip: +formData.zip,
				city: formData.city,
				country: formData.country,
				token: token,
			})
		);

		if (resp.type === "NEW_PROJECT_ERROR") {
			// could not create new project. return don't redirect, snackbar should provide user feedback
			return;
		}

		resetFormData();
		history.push(`/user`);
	};

	return (
		<div>
			<Container component="main" maxWidth="xs">
				<div className={classes.paper}>
					<CssBaseline />
					<Typography component="h1" variant="h5">
						Add New Project
					</Typography>
					<form className={classes.form} noValidate>
						<BasicProjectForm
							formData={formData}
							handleChange={handleChange}
						/>
						<Button
							type="submit"
							fullWidth
							variant="contained"
							color="primary"
							className={classes.submit}
							onClick={handleSubmit}
						>
							Create New Project
						</Button>
					</form>
				</div>
			</Container>
		</div>
	);
};

export default NewProjectForm;
