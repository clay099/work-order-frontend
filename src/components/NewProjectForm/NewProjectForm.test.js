import React from "react";
import NewProjectForm from "./NewProjectForm";
import { render } from "../../testHelper/testHelper";
console.error = jest.fn();

const params = {
	token: "token",
};

describe("NewProjectForm Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<NewProjectForm {...params} />);
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(<NewProjectForm {...params} />);
		expect(asFragment).toMatchSnapshot();
	});

	test("value to be in the document", () => {
		const { getByText } = render(<NewProjectForm {...params} />);
		const value = getByText("Create New Project");
		expect(value).toBeInTheDocument();
	});
});
