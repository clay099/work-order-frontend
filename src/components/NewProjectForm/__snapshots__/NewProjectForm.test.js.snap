// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProjectForm Component matches snapshot 1`] = `[Function]`;
